File: Rust/algo/src/sort/selection.rs
pub fn selection_sort<T:Ord + Copy>(arr: &mut [T]) {
    for i in 0..arr.len()-1 {
        let mut sel = i;
        for j in i+1..arr.len() {
            if arr[j] < arr[sel] {
                sel = j;
            }
        }
        (arr[i], arr[sel]) = (arr[sel], arr[i])
    }
}

File: C/algorithms/sort/selection.c
void selection_sort(int arr[], int n)
{
    int i, j, smallest;
    for (i=0; i<n-1; i++){
        smallest = i;

        for (j=i+1; j<n; j++){
            if (arr[j]<arr[smallest]){
                smallest = j;
            }
        }
        if(smallest != i){
            int temp = arr[smallest];
            arr[smallest] = arr[i];
            arr[i] = temp;
        }
    }
}